src/extension/gogoduck/src/test/java/au/org/emii/gogoduck/worker/GoGoDuckModuleTest.java

```diff
@@ -0,0 +1,72 @@
+package au.org.emii.gogoduck.worker;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import au.org.emii.gogoduck.worker.GoGoDuck.UserLog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ucar.nc2.Attribute;
+import ucar.nc2.NetcdfFile;
+import ucar.nc2.NetcdfFileWriter;
+
+public class GoGoDuckModuleTest {
+    public GoGoDuckModule ggdm = null;
+
+    @Before
+    public void beforeEach() {
+        ggdm = new GoGoDuckModule();
+        ggdm.init("", "", "TIME,1,2;LONGITUDE,2,3", null);
+    }
+
+    @Test
+    public void testGetSubsetParameters() throws Exception {
+        // Make sure it removes `TIME` subset parameter
+        assertFalse(ggdm.getSubsetParameters().containsKey("TIME"));
+
+        // But did that only on a copy of the subset parameters
+        Field privateSubset = GoGoDuckModule.class.getDeclaredField("subset");
+        privateSubset.setAccessible(true);
+        SubsetParameters sp = (SubsetParameters) privateSubset.get(ggdm);
+        assertTrue(sp.containsKey("TIME"));
+    }
+
+    public class GoGoDuckModule_test extends GoGoDuckModule {
+        @Override
+        protected List<Attribute> getGlobalAttributesToUpdate(NetcdfFile nc) {
+            List<Attribute> newAttributeList = new ArrayList<Attribute>();
+            newAttributeList.add(new Attribute("test_attribute", "test_value"));
+            return newAttributeList;
+        }
+    }
+
+    @Test
+    public void testUpdateMetadata() throws Exception {
+        File sampleNetcdf = new File("src/test/resources/file.nc").getAbsoluteFile();
```

Comment:
> I would say not, but in production code, you would probably want to use a FileOutputStream (if expecting to deal with such large amounts of data). In tests, use a ByteOutputStream (since minimal data, faster, cleaner).

:+1: Agreed.